### ISSUE TITLE
[Search v2.5] Remove special handling of policyID after navigation is merged

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#53480]
We need to remove [this code](https://github.com/Expensify/App/pull/53198/files#diff-3e88ec721348df7dd294340601a56f7af194ca49fe8ad06248e1ba6c895c070dR140-R142) once [this navigation PR](https://github.com/Expensify/App/pull/49539) is merged.

@Kicu

**Comment by Kicu** [comment]
Thanks for the issue.

Some extra context:
PolicyID (aka `workspaceID`) has a lot of special handling, as it comes from URLs but is also saved to `workspaceContext`. To avoid having multiple ways of accessing it, we try to access it from URL/route params whenever possible, and handle it on navigation actions.
In this specific case instead of overwriting workspace id, we can simply check if there is existing id during navigation and handle it.
That is already working in #49539

**Comment by Kicu** [comment]
also please assign me, so I won't forget about this 🙏

**Comment by luacmartins** [comment]
On hold

**Comment by luacmartins** [comment]
@Kicu the holding PR has been merged. Can we resume work on this one? This is lower priority than the current report projects though

**Comment by Kicu** [comment]
Yes, but its a small thing. I would prefer to just do it in my own time, inbetween Reports stuff when I find some time. 
So maybe in the latter half of this week I will take a look

**Comment by luacmartins** [comment]
Yea, I agree the priority is low

**Comment by Kicu** [comment]
@luacmartins I have recently synced with @WojtekBoman about incoming changes in navigation. Not sure how in the loop you are, but we are planning to drop workspace switcher completely, and have `policyID` behave like any other filter.

Because of this I think we can completely close this task, as after policy changes code related to this issue will simply disappear.

**Comment by luacmartins** [comment]
Ah that makes sense. Agree that we can close this then. Thanks!
